src/aiida/tools/dumping/utils.py

```diff
@@ -0,0 +1,72 @@
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Utility functions for dumping features."""
+
+from __future__ import annotations
+
+import logging
+import shutil
+from pathlib import Path
+
+__all__ = ['validate_make_dump_path']
+
+logger = logging.getLogger(__name__)
+
+
+def validate_make_dump_path(
+    path_to_validate: Path,
+    overwrite: bool = False,
+    incremental: bool = True,
+    safeguard_file: str = '.aiida_node_metadata.yaml',
+    enforce_safeguard: bool = True,
+) -> Path:
+    """Create default dumping directory for a given process node and return it as absolute path.
+
+    :param validate_path: Path to validate for dumping.
+    :param safeguard_file: Dumping-specific file to avoid deleting wrong directory.
+        Default: `.aiida_node_metadata.yaml`
+    :return: The absolute created dump path.
+    """
```

Comment:
Yeah, that's the reason. While I was developing, I still had some bug in the code, and actually deleted a bunch of files which I didn't intend to delete. So I added this as a safeguard, just in case. With a correct implementation, it shouldn't really matter anymore, though, I think it's still good to keep, as users might accidentally tab-complete on the command line, and select a directory, which would then get deleted. I'll add some more text.